lib/charms/opensearch/v0/helper_security.py

```diff
@@ -117,3 +126,232 @@ def to_pkcs8(private_key: str, password: Optional[str] = None) -> str:
     finally:
         os.unlink(tmp_key.name)
         os.unlink(tmp_pkcs8_key.name)
+
+
+def split_ca_chain(pem_content: str) -> list[str]:
+    """Split PEM chain into individual certificates."""
+    end_cert_marker = "-----END CERTIFICATE-----"
+    parts = [part.strip() for part in pem_content.split(end_cert_marker) if part.strip()]
+    return [f"{part}\n{end_cert_marker}" for part in parts]
+
+
+def store_ca(
+    alias: str, store_pwd: str, store_path: str, ca: str, keep_previous: bool = True
+) -> bool:
+    """Add new CA cert to trust store."""
+    # This loop and split are to handle when the CA is a chain with intermediate certs
+    certs = list(reversed(split_ca_chain(ca)))
+
+    for index in range(len(certs)):
+        if keep_previous:
+            cmd = f"{KEYTOOL} -changealias -alias {alias}-{index} -destalias old-{alias}-{index} -keystore {store_path} -storetype PKCS12"
```

Comment:
Question: What will happen to the stored old ca ? Are we just going to store them in case of need or there is a different process that will leverage them ? If that is the case, I think the name of the alias should be normalized in a separate constant instead of specifying "old-{alias}-{index}"

Reply:
I got my answer when reading the tls module, but still think that if we are using the function from another module we would still need to be aware of this. 
I would personally have a constant OLD_CA_PREFIX="old" . But this is very minor